sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -3527,13 +3527,35 @@ class Analyzer(override val catalogManager: CatalogManager)
    * Rule to mostly resolve, normalize and rewrite column names based on case sensitivity
    * for alter table commands.
    */
-  object ResolveFieldNames extends Rule[LogicalPlan] {
+  object ResolveAlterTableCommands extends Rule[LogicalPlan] {
     def apply(plan: LogicalPlan): LogicalPlan = plan.resolveOperatorsUp {
       case a: AlterTableCommand if a.table.resolved =>
-        a.transformExpressions {
+        val table = a.table.asInstanceOf[ResolvedTable]
+        val transformed = a.transformExpressions {
           case u: UnresolvedFieldName =>
-            val table = a.table.asInstanceOf[ResolvedTable]
-            resolveFieldNames(table.schema, u.name).map(ResolvedFieldName(_)).getOrElse(u)
+            resolveFieldNames(table.schema, u.name).getOrElse(u)
+          case u: UnresolvedFieldPosition => u.position match {
+            case after: After =>
+              resolveFieldNames(table.schema, u.fieldName.init :+ after.column())
```

Comment:
Do we need to put `fieldName` in `UnresolvedFieldPosition`? We can easily get it via `AlterTableAlterColumn.column.name`

Reply:
https://github.com/apache/spark/pull/33213